### PR TITLE
Add a new script to download sample datasets

### PR DESCRIPTION
**Pull request by tats-u**
I created a new script to download sample datasets (e.g. for VCC 2016).  `/example/download_vcc2016data.py` is difficult to maintain and adapt to other datasets and must be deprecated.

New script adopts YAML-based configuration files to support other datasets than what I preset much more easily.  The presets are for not only VCC 2016 but also VCC 2018.

### CODE REVIEW

**Comment by k2kobayashi** [comment]
When I try to use this script at first time, the scripts does not work due to following errors.
This problem may be caused by the overwrite exception without option.

```
./download_sample_data.py downloader_conf/vcc2018.yml
Downloading training data from https://datashare.is.ed.ac.uk/bitstream/handle/10283/3061/vcc2018_database_training.zip?sequence=2&isAllowed=y ...
^@^@^@^@^@^@^@^@Unpack: /var/folders/q9/0fl205qj5d7bfw_sw32jhz_c0000gn/T/tmp7z0ug6vq/vcc2018_database_training.zip
Move: VCC2SF1
Move: VCC2TM2
Move: VCC2TF1
Move: VCC2SM2
Move: VCC2SM4
Move: VCC2SM3
Move: VCC2SF3
Move: VCC2SF4
Move: VCC2SF2
Move: VCC2TM1
Move: VCC2TF2
Move: VCC2SM1
Downloading evaluation data from https://datashare.is.ed.ac.uk/bitstream/handle/10283/3061/vcc2018_database_evaluation.zip?sequence=3&isAllowed=y ...
^@^@^@Unpack: /var/folders/q9/0fl205qj5d7bfw_sw32jhz_c0000gn/T/tmpdg99x83t/vcc2018_database_evaluation.zip
Move: VCC2SF1
Traceback (most recent call last):
  File "./download_sample_data.py", line 511, in <module>
    wav_root_dir
  File "./download_sample_data.py", line 492, in download
    file.download(dest)
  File "./download_sample_data.py", line 374, in download
    working_dir / self.audio_root_relative, dest_root
  File "./download_sample_data.py", line 395, in _move_all_audio
    self._move_audio_in_dir(directory, dest_dir)
  File "./download_sample_data.py", line 410, in _move_audio_in_dir
    os.makedirs(str(dest), exist_ok=self.behavior.force)
  File "/Users/Kazuhiro/.pyenv/versions/3.6.5/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: 'data/wav/VCC2SF1'
```

File: example/download_sample_data.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+import urllib.request
+from fnmatch import fnmatchcase
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import yaml
+from docopt import docopt
+
+
+class Behaivior:
```
[the remark this answers is not in the document]
Comment (tats-u) [reply example/download_sample_data.py]: I'll change it to `UserOption`

**Comment by tats-u** [comment]
What should I do with behaviors with and without `-f` option?

File: example/download_sample_data.py [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+
+Usage:
+    download_sample_data.py <config> [-h] [-q] [-f] [-m]
+
+Parameters:
+    <config>        The path of configuration file
+    -h, --help      Show this help and exit
+    -q, --quiet     Don't show any messages about progress
+    -f, --force     Overwrite existing directories and files
+    -m, --minimal   Exit when there are some directories in data/wav
```
[the remark this answers is not in the document]
Comment (tats-u) [reply example/download_sample_data.py]: I'll omit it.

**Comment by k2kobayashi** [comment]
@tats-u 
For the ```-f``` option, it should overwrite not the target directory to be stored but the storing wave file.

**Comment by tats-u** [comment]
Does it mean directories with the same names as those in archives are allowed to exist regardless of `-f` flag?

**Comment by k2kobayashi** [comment]
Yes. It should be allowed to remove some wav files and re-downloaded the removed wav files via the download script.

**Comment by k2kobayashi** [comment]
Thanks @tats-u . We merge it.

**Pull request merged**